tests/aws/services/events/conftest.py

```diff
@@ -392,3 +392,59 @@ def collect_events() -> None:
     retry(collect_events, retries=retries, sleep=0.01)
 
     return events
+
+
+@pytest.fixture
+def logs_create_log_group(aws_client):
+    log_group_names = []
+
+    def _create_log_group(name: str = None) -> str:
+        if not name:
+            name = f"test-log-group-{short_uid()}"
+
+        aws_client.logs.create_log_group(logGroupName=name)
```

Comment:
@maxhoheiser I learned while testing pipes that log groups are not created instantly at AWS. Even worse was that a subsequent `describe_log_groups` call failed in this example:

```py
log_group_name = f"test-log-group-{short_uid()}"
aws_client.logs.create_log_group(logGroupName=log_group_name)

def _log_group_created():
    describe_log_groups_response = aws_client.logs.describe_log_groups(
        logGroupNamePattern=log_group_name
    )
    log_groups = describe_log_groups_response.get("logGroups", [])
    return len(log_groups) > 0

# create_log_group does not happen instantly on AWS
assert poll_condition(_log_group_created, timeout=5, interval=0.5)

describe_log_groups_response = aws_client.logs.describe_log_groups(
    logGroupNamePattern=log_group_name
)
log_group_arn = describe_log_groups_response["logGroups"][0]["arn"]
cleanups.append(lambda: aws_client.logs.delete_log_group(logGroupName=log_group_name))
```

with the following error:

```
>       log_group_arn = describe_log_groups_response["logGroups"][0]["arn"]
E       IndexError: list index out of range
```

I ended up returning the `log_group_arn` through the helper. Ideally, we could reuse a reliable fixture.


👉  It might be worth adding a comment to the fixture or implementing a waiter.